src/Rfc1035StubDnsResolver.php

```diff
@@ -190,13 +190,15 @@ public function resolve(string $name, ?int $typeRestriction = null, ?Cancellatio
                     }
 
                     return \array_merge(...$records);
-                } catch (MissingDnsRecordException) {
-                    $alias = $this->searchForAliasRecord($searchName, $cancellation);
-                    if ($alias !== null) {
-                        $name = $alias;
-                    }
-                    continue;
                 } catch (DnsException $e) {
+                    if ($e instanceof MissingDnsRecordException) {
```

Comment:
Why instanceof instead of a separate catch block?

Reply:
If no alias record is found, the code falls-through to the check on search index. A separate block would have to repeat that code.